pkg/assets/apiserver/asset_apiserver.go

```diff
@@ -104,13 +118,57 @@ func NewAssetServerConfig(assetConfig v1.WebConsoleConfiguration) (*AssetServerC
 }
 
 // Complete fills in any fields not set that are required to have valid data. It's mutating the receiver.
-func (c *AssetServerConfig) Complete() completedConfig {
+func (c *AssetServerConfig) Complete() (completedConfig, error) {
 	cfg := completedConfig{
 		c.GenericConfig.Complete(),
+		c.GenericConfig.ClientConfig,
 		&c.ExtraConfig,
 	}
 
-	return cfg
+	// In order to build the asset server, we need information that we can only retrieve from the master.
+	// We do this once during construction at the moment, but a clever person could set up a watch or poll technique
+	// to dynamically reload these bits of config without having the pod restart.
+	restClient, err := kubernetes.NewForConfig(c.GenericConfig.ClientConfig)
+	if err != nil {
+		return completedConfig{}, err
+	}
+	if len(cfg.ExtraConfig.KubeVersion) == 0 {
+		resultBytes, err := restClient.RESTClient().Get().AbsPath("/version").Do().Raw()
+		if err != nil {
+			return completedConfig{}, err
+		}
+		kubeVersion := &version.Info{}
+		if err := json.Unmarshal(resultBytes, kubeVersion); err != nil {
+			return completedConfig{}, err
+		}
+		cfg.ExtraConfig.KubeVersion = kubeVersion.GitVersion
+	}
+	if len(cfg.ExtraConfig.OpenShiftVersion) == 0 {
+		resultBytes, err := restClient.RESTClient().Get().AbsPath("/version/openshift").Do().Raw()
+		if err != nil {
+			return completedConfig{}, err
+		}
+		openshiftVersion := &version.Info{}
+		if err := json.Unmarshal(resultBytes, openshiftVersion); err != nil {
+			return completedConfig{}, err
+		}
+		cfg.ExtraConfig.OpenShiftVersion = openshiftVersion.GitVersion
+	}
+	if cfg.ExtraConfig.TemplateServiceBrokerEnabled == nil {
+		enabled := false
+		_, err := restClient.RESTClient().Get().AbsPath("/apis/servicecatalog.k8s.io/v1beta1/clusterservicebrokers/template-service-broker").Do().Raw()
```

Comment:
Will this work correctly if service catalog is not installed?

Reply:
> Will this work correctly if service catalog is not installed?

Yes, it will return a 404 and simply set itself to "enabled=false"